### DISCUSSION TITLE
nginx-ingress controller :1.12.4

### DISCUSSION THREAD

**Discussion by dbsom2018batch** [nginx/kubernetes-ingress#2609] [Q&A] [answered]
while running the make debian-image-plus for the version 1.12.4 in the local i got an error

RUN CGO_ENABLED=0 GO111MODULE=on GOOS=linux GOARCH=amd64 go build -trimpath -ldflags "-s -w -X main.version=v1.12.4-SNAPSHOT-addceb8 -X main.commit=addceb86fb4cfa225696a1f88f649d625e5e8857 -X main.date=2022-04-14T11:50:13Z" -o /nginx-ingress:
#19 11.32 # sigs.k8s.io/json/internal/golang/encoding/json
#19 11.32 /go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
#19 11.32 /go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
------
executor failed running [/bin/sh -c CGO_ENABLED=0 GO111MODULE=on GOOS=linux GOARCH=$TARGETARCH go build -trimpath -ldflags "-s -w -X main.version=${IC_VERSION} -X main.commit=${GIT_COMMIT} -X main.date=${DATE}" -o /nginx-ingress]: exit code: 2
make: *** [Makefile:73: debian-image-plus] Error 1

But it was working fine in 1.12.3 version, does anyone have any input or suggestion how can i fix ?

**Comment by lucacome** [comment, marked as answer]
Hi @dbsom2018batch 

are you running `make debian-image-plus` with `TARGET=container`? It looks like we have a bug and this line https://github.com/nginxinc/kubernetes-ingress/blob/v1.12.4/build/Dockerfile#L250 should be `FROM golang:1.17-alpine AS builder`. If you're able to modify that line, it will build successfully.

Alternatively, if you have a local go environment with a go version >= 1.17 you build with `make debian-image-plus` with `TARGET=local`.

I'll open a PR to fix this.

**Comment by dbsom2018batch** [comment]
@lucacome  Thank you so much for the quick response, it would be great if you can fix it as i am using team-city CI/CD tool and want 1.12.4 image push into my client azure private repository.

**Reply by lucacome** [reply]
@dbsom2018batch do you clone the branch or the tag? The fix should go in pretty soon, but I'm not sure when we'll be able to make a new release.

Also in case you didn't know, we have pre-built images in our repo that you can pull if that's an option for you, see the doc [here](https://docs.nginx.com/nginx-ingress-controller/installation/pulling-ingress-controller-image/)

**Reply by dbsom2018batch** [reply]
Thank you @lucacome
